flink-connector-pulsar-e2e-tests/pom.xml

```diff
@@ -23,7 +23,7 @@ under the License.
 	<parent>
 		<groupId>org.apache.flink</groupId>
 		<artifactId>flink-connector-pulsar-parent</artifactId>
-		<version>4.1-SNAPSHOT</version>
+		<version>4.2-SNAPSHOT</version>
```

Comment:
Just thoughts out loud
since now it is maven 3.8.6
I'm curious whether we could use `revision` for versions and keep it in one place only 
https://maven.apache.org/maven-ci-friendly.html

Reply:
@snuyanzin This may need a mailing list discussion. I personally support to use `4.2.0-SNAPSHOT` but the whole Flink repos use `X.Y-SNAPSHOT` for a long time and I don't know if other scripts depend on this format.

cc @zentol 

For this patch, I tend to keep the current flavor.